crates/forge/bin/cmd/test/mod.rs

```diff
@@ -475,9 +475,9 @@ impl TestArgs {
 
                 // Identify addresses and decode traces.
                 let mut decoded_traces = Vec::with_capacity(result.traces.len());
-                for (kind, arena) in &result.traces {
+                for (kind, mut arena) in result.traces.clone() {
```

Comment:
here it should be &mut and decode separately, so that below you don't have to re-decode again in gas report

Reply:
@DaniPopes fixed the `&mut`, not sure what you mean by decoding separately, do you mean moving 

```rust
   decoder.prefetch_signatures(arena.nodes()).await;
    decoder.extend_traces(arena.nodes_mut()).await;
```

out from `render_trace_arena`?

Or creating a `decoded_arenas` array to collect the decoded traces to be re-used by the gas report below?

Reply:
The first, `render_trace_arena` should just be `&Arena -> String`